### PR TITLE
Add GUI save button and listeners to connect to write_loop function

### PR DESCRIPTION
**Pull request by sd-owens**
- Add GUI buttons and listeners to open a SAVE file dialog box allowing for entry of file name or selecting of file to overwrite.
- Connected GUI button to provide filepath name to write_loop function for writing JSON files.

### CODE REVIEW

File: audnauseum/gui/ui.py [hunk 1]
```diff
@@ -120,6 +128,18 @@ def open_file_dialog(ui) -> str:
...
 
 
+# Modified from source code example: https://pythonspot.com/pyqt5-file-dialog/
+
+def save_file_dialog(ui) -> str:
+    options = QFileDialog.Options()
+    options |= QFileDialog.DontUseNativeDialog
+
+    file_path, _ = QFileDialog.getSaveFileName(
+        ui, "QFileDialog.getSaveFileName()", "", "Loops Files (*.json)", options=options)
```
Comment (davidmikulis) [inline audnauseum/gui/ui.py:138]: Suggestion - update the title of the dialog to something more user-friendly and open the dialog to the "resources/json" folder.
```suggestion
        ui, "Save a Loop File", "./resources/json", "Loops Files (*.json)", options=options)
```

**Review by davidmikulis — APPROVED** [review]
Looks good to me! I added one suggestion that's cosmetic only.

Thanks for also fixing the *args thing for the UI functions.

File: audnauseum/gui/ui.py (hunk 1 above)
Comment (sd-owens) [reply audnauseum/gui/ui.py:138]: Agreed, I will fix that in the load and save dialogs

**Review by robomoto — APPROVED** [review]
Looks great!

**Pull request merged**